### PR TITLE
fix(ingest/bigquery) - Lineage edges use datetime with timezone

### PR DESCRIPTION
**Pull request by treff7es**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py [hunk 1]
```diff
@@ -487,7 +487,11 @@ def get_views_for_dataset(
             BigqueryView(
                 name=table.table_name,
                 created=table.created,
-                last_altered=table.get("last_altered", table.created),
+                last_altered=datetime.fromtimestamp(
+                    table.get("last_altered") / 1000, tz=timezone.utc
+                )
+                if table.get("last_altered") is not None
+                else table.created,
```
Comment (asikowitz) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py:494]: So based on https://cloud.google.com/bigquery/docs/information-schema-tables#schema `creation_time` is a TIMESTAMP just like `last_altered` (at least in the [PARTITIONS view](https://cloud.google.com/bigquery/docs/information-schema-partitions#schema)). How come we do datetime conversion for one but not the other? Am I missing something?

File: metadata-ingestion/tests/unit/test_bigquery_lineage.py [hunk 2]
```diff
@@ -90,3 +98,79 @@ def test_create_statement_with_multiple_table():
...
     assert "my_project_2.my_dataset_2.sometable2" == tables[1].get_table_name()
+
+
+def test_lineage_tables():
+    config = BigQueryV2Config()
+    report = BigQueryV2Report()
+    extractor: BigqueryLineageExtractor = BigqueryLineageExtractor(config, report)
+    lineage_entries: List[QueryEvent] = [
+        QueryEvent(
+            timestamp=datetime.datetime.now(),
```
Comment (asikowitz) [inline metadata-ingestion/tests/unit/test_bigquery_lineage.py:109]: Do these timestamps come timezone from the bigquery client library?

**Review by asikowitz — COMMENTED** [review]
I'm confused why `last_altered` is a timestamp but `created` is a datetime. Otherwise looks good to me; we should do a datetime timezone audit afterwards

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py (hunk 1 above)
Comment (asikowitz) [reply metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py:494]: Seems like it just works this way :/ guess it's a fun features of the `__TABLES__` table...

**Review by asikowitz — APPROVED** [review]

**Comment by hsheth2** [comment]
@treff7es looks like there's real test failures on this one

**Pull request merged**